index.ts

```diff
@@ -42,42 +43,46 @@ if (process.env.CI == "true") {
     const launchArgs = process.argv;
     for (let i = 0; i < launchArgs.length; i++) {
         if (launchArgs[i] === "--owner_source") {
-            owner_source = launchArgs[i + 1];
+            ownerSource = launchArgs[i + 1];
         } else if (launchArgs[i] === "--repo_source") {
-            repo_source = launchArgs[i + 1];
+            repoSource = launchArgs[i + 1];
         } else if (launchArgs[i] === "--owner_target") {
-            owner_target = launchArgs[i + 1];
+            ownerTarget = launchArgs[i + 1];
         } else if (launchArgs[i] === "--repo_target") {
-            repo_target = launchArgs[i + 1];
+            repoTarget = launchArgs[i + 1];
         } else if (launchArgs[i] === "--github_token") {
-            GITHUB_TOKEN = launchArgs[i + 1];
+            githubToken = launchArgs[i + 1];
+        } else if (launchArgs[i] === "--github_token_source") {
+            githubTokenSource = launchArgs[i + 1];
+        } else if (launchArgs[i] === "--github_token_target") {
+            githubTokenTarget = launchArgs[i + 1];
         }
     }
 }
 
-// Init octokit
-const octokit = new Octokit({
-    auth: GITHUB_TOKEN,
-    // TODO: add ghes IP support here, or use github.octokit
+const octokitSource = new Octokit({
+    auth: githubTokenSource,
+    // TODO: add GHES IP support here, or use github.octokit
+});
+
+const octokitTarget = githubTokenSource == githubTokenTarget ? octokitSource : new Octokit({
+    auth: githubTokenTarget,
+    // TODO: add GHES IP support here, or use github.octokit
 });
```

Comment:
why not create those in the constructor of the class?
If you need separate access to octokit somewhere else, you could still access it if you have the initialised class.
Atm all variables of the GitHub class are public by default.

Reply:
Good point, refactored it.